File: 3sem/4lab/Banks/Exceptions/InvalidMoneyGapException.cs
namespace Banks.Exceptions;

public class InvalidMoneyGapException : BanksDomainException
{
    private InvalidMoneyGapException(string? message)
        : base(message) { }

    public static InvalidMoneyGapException OnInvalidMoneyAmount()
        => new InvalidMoneyGapException("Money amount must be positive");

    public static InvalidMoneyGapException OnInvalidInterestRate()
        => new InvalidMoneyGapException($"Interest rate must be in [0; 1]");

    public static InvalidMoneyGapException OnFromSumGreaterThanTo()
        => new InvalidMoneyGapException($"'From' sum must be lower than 'To' sum");
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Exceptions/InvalidMoneyGapException.java
package banksRemake.domain.exceptions;

public class InvalidMoneyGapException extends BanksRemakeException {
    private InvalidMoneyGapException(String message) {
        super(message);
    }

    public static InvalidMoneyGapException onInvalidMoneyAmount() {
        return new InvalidMoneyGapException("Money amount must be positive");
    }

    public static InvalidMoneyGapException onInvalidInterestRate() {
        return new InvalidMoneyGapException("Interest rate must be in [0; 1]");
    }

    public static InvalidMoneyGapException onFromSumGreaterThanTo(){
        return new InvalidMoneyGapException("'From' sum must be lower than 'To' sum");
    }
}